hotsos/core/config.py

```diff
@@ -159,9 +160,30 @@ def name(self):
 class RegisteredOpts(UserDict):
 
     def __init__(self, *optgroups):
+        self.optsgroups = []
         self.data = {}
-        for ogroup in optgroups:
-            self.data.update(ogroup())
+        for optgroup in optgroups:
+            optgroup = optgroup()
+            self.optsgroups.append(optgroup)
+            a = [name.lower() for name in self.data.keys()]
+            b = [name.lower() for name in optgroup.keys()]
```

Comment:
Could add a test for groups/names being duplicated based on case, but wouldn't hold it over that.

Reply:
done.